api/tests/unit/features/test_unit_features_views.py

```diff
@@ -2588,6 +2589,10 @@ def test_list_features_n_plus_1(
 ) -> None:
     # Given
     with_project_permissions([VIEW_PROJECT])
+    num_queries = 16
+
+    if settings.IS_RBAC_INSTALLED:  # pragma: no cover
+        num_queries += 1
```

Comment:
I'd rather parametrise the test and mock the setting explicitly.

Reply:
I'd rather not mock the settings and cover both the cases

Reply:
> I'd rather not mock the settings

I can't understand what is wrong with mocking the settings here?

> cover both the cases

We can still keep it DRY and cover both paths by same test code if we parametrise the mock.

On the contrary, the current approach requires to setup the test run twice with different setting modules to cover both paths (as evident by the `pragma: no cover` comment).

Reply:
I guess what Gagan is trying to achieve here is a generic solution that works for both (one which includes all the private packages) and one which doesn't. 

I guess with @khvn26 's approach, we could add a `pytest.mark.skip` perhaps?

Reply:
A skip mark along with a separate test would be preferable, agreed.

Reply:
I'd prefer to not duplicate and maintain two tests that pretty much test the same thing, but let me know if you feel strongly about this @khvn26

Reply:
So, I guess to summarse, we have a few options here: 

 1. Keep it as it is, although I agree with @khvn26 that the pragma no cover is a bit of a code smell here. 
 2. Write 2 separate tests, one skipped if `settings.IS_RBAC_INSTALLED = False`. Note that we wouldn't necessarily need to maintain 2 separate tests here, right, the test code could all be in a central, private function, then we just call that from each separate test function. 

I think there probably is a third option (or set of options) where we do mock out the settings instead, right? And let the tests from the rbac module itself handle the code paths that we're trying to test here. I'm not sure how feasible this is?